### PR TITLE
MULE-13891: Non repeatable streams are not managed to guarantee closi…

### PR DESCRIPTION
**Pull request by marianogonzalez**
…ng them

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java [hunk 1]
```diff
@@ -124,6 +129,19 @@ public CursorProvider manage(CursorProvider provider, CoreEvent creatorEvent) {
    * {@inheritDoc}
    */
   @Override
+  public void manage(InputStream stream, CoreEvent creatorEvent) {
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: Why do you need to pass event, doesn't EventContext suffice?

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java [hunk 2]
```diff
@@ -199,6 +201,26 @@ public void streamProviderSerialization() throws Exception {
...
+  public void nonRepeatableStreamClosesAsync() throws Exception {
+    InputStream stream = (InputStream) flowRunner("toNonRepeatableStream").keepStreamsOpen()
+        .withPayload(data).run().getMessage().getPayload().getValue();
+
+    byte[] bytes = new byte[data.length()];
+    assertThat(stream.read(bytes), is(data.length()));
+    assertThat(stream.read(), is(-1));
+    assertThat(new String(bytes), equalTo(data));
+  }
+
```
Comment (dfeist) [inline modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java]: Need to test the error case too.

Comment (marianogonzalez) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java]: which error case?

File: core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java (hunk 1 above)
Comment (marianogonzalez) [reply core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: i pass the event mainly for consistency with the other manage method which is already there and receives the full event. A very secondary `future proofing` point could also be made. By passing the entire event the API is still good if we ever need something else than the context. But that argument pales compared to the consistency one

Comment (dfeist) [reply core/src/main/java/org/mule/runtime/core/api/streaming/DefaultStreamingManager.java]: ok. thought it was that, just checking,

File: modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java (hunk 2 above)
Comment (dfeist) [reply modules/extensions-spring-support/src/test/java/org/mule/test/module/extension/streaming/BytesStreamingExtensionTestCase.java]: The case where an error is thrown in a flow and error is handled by an error-handler. This is a different code path and its critical that streams are also closed in this case, so it should really be covered with tests

**Pull request merged**